app/server.go

```diff
@@ -827,6 +838,59 @@ func doSessionCleanup(s *Server) {
 	s.Store.Session().Cleanup(model.GetMillis(), SESSIONS_CLEANUP_BATCH_SIZE)
 }
 
+func doStoreAndCheckNumberOfActiveUsersWarnMetricStatus(s *Server) {
+	// uncomment this if we want to have this check only for TE
+	// license := s.License()
+	// if license != nil {
+	// 	mlog.Debug("License is present, skip this check")
+	// 	return
+	// }
+
+	props, err := s.Store.System().Get()
+	if err != nil {
+		mlog.Error("Error to get system settings.", mlog.Err(err))
+		return
+	}
+
+	if props[model.SYSTEM_NUMBER_OF_ACTIVE_USERS_WARN_METRIC] != "" {
+		val, err := strconv.ParseInt(props[model.SYSTEM_NUMBER_OF_ACTIVE_USERS_WARN_METRIC], 10, 64)
+		if err != nil {
+			mlog.Error("Number Of Active Users Warn Metric", mlog.String("Invalid entry value stored", model.SYSTEM_NUMBER_OF_ACTIVE_USERS_WARN_METRIC))
+		}
+		if val == -1 {
+			mlog.Info("Exceeding the number of active users metric limit has been already acknowledged")
+			return
+		}
+	} else {
+		mlog.Debug("Number Of Active Users Metric", mlog.String("Cannot find metric in store", model.SYSTEM_NUMBER_OF_ACTIVE_USERS_WARN_METRIC))
+	}
+
+	//change MONTH_MILLISECONDS to a different value if we want to capture active users for a different period
+	noActiveUsers, err := s.Store.User().AnalyticsActiveCount(MONTH_MILLISECONDS, model.UserCountOptions{IncludeBotAccounts: false, IncludeDeleted: false})
+	if err != nil {
+		mlog.Error("Error to get active registered users.", mlog.Err(err))
+	}
+
+	mlog.Info("Number of active users", mlog.Int64("value", noActiveUsers))
+
+	if err = s.Store.System().SaveOrUpdate(&model.System{Name: model.SYSTEM_NUMBER_OF_ACTIVE_USERS_WARN_METRIC, Value: strconv.FormatInt(noActiveUsers, 10)}); err != nil {
+		mlog.Error("Unable to write to database.", mlog.Err(err))
+		return
+	}
+
+	if noActiveUsers > model.NUMBER_OF_ACTIVE_USERS_WARN_METRIC_LIMIT {
+		mlog.Info("Number of active users is greater than limit")
+		message := model.NewWebSocketEvent(model.WEBSOCKET_WARN_METRICS_STATUS, "", "", "", nil)
```

Comment:
Also certain why we need to send a websocket event, isn't posting the message good enough?

Reply:
Sorry, I think I'm getting this confused with the other ticket.  Guessing this is needed for the banner bar to pop up.

Reply:
yes, both bar and DM are handled here.